examples/terraform/gcp/gke-autopilot.tf

```diff
@@ -0,0 +1,59 @@
+# VPC
+resource "google_compute_network" "gke_autopilot_network" {
+  name                    = "gke-autopilot-network"
+  project                 = local.project_id
+  auto_create_subnetworks = false
+}
+
+# Subnet
+resource "google_compute_subnetwork" "gke_autopilot_subnet" {
+  name          = "gke-autopilot-subnet"
+  project       = local.project_id
+  ip_cidr_range = "10.5.0.0/16"
+  region        = var.region
+  network       = google_compute_network.gke_autopilot_network.id
+}
+
+# GKE Autopilot private cluster
+resource "google_container_cluster" "default" {
+  name     = "gke-autopilot"
+  project  = local.project_id
+  location = var.region
+
+  # Private clusters use nodes that do not have external IP addresses.
+  # This means that clients on the internet cannot connect to the IP addresses of the nodes.
+  private_cluster_config {
+    enable_private_nodes    = true
+    enable_private_endpoint = true
+    # Control plane nodes are not accessible globally
+    master_global_access_config {
+      enabled = false
+    }
+  }
+
+  # At this point, these are the only IP addresses that have access to the control plane:
+  #   - The primary range of my-subnet.
+  #   - The secondary range my-pods.
```

Comment:
We need to clarify a bit better what `my-subnet` and `my-pods` networks are for. Pointing to the terraform resource, or explaining better what they will be used for.